### ISSUE TITLE
Error in Appender: Negative transfer amount after block 2871360

### ISSUE THREAD

**Issue by jaydp17** [wavesplatform/Waves#3592]
**Describe the bug**
All of my nodes are crashing with the following error

```
ERROR [rx-extension-loader-42] c.w.Application - Error in Appender
com.wavesplatform.lang.v1.evaluator.RejectException: Negative transfer amount = -15220
	at com.wavesplatform.state.diffs.invoke.InvokeDiffsCommon$.$anonfun$checkScriptResultFields$1(InvokeDiffsCommon.scala:666)
	at com.wavesplatform.state.diffs.invoke.InvokeDiffsCommon$.$anonfun$checkScriptResultFields$1$adapted(InvokeDiffsCommon.scala:663)
	at scala.collection.immutable.List.foreach(List.scala:333)
	at com.wavesplatform.state.diffs.invoke.InvokeDiffsCommon$.checkScriptResultFields(InvokeDiffsCommon.scala:663)
	at com.wavesplatform.state.diffs.invoke.InvokeScriptTransactionDiff$.$anonfun$evaluateV2$3(InvokeScriptTransactionDiff.scala:303)
	at scala.util.Either.map(Either.scala:382)
	at com.wavesplatform.state.diffs.invoke.InvokeScriptTransactionDiff$.evaluateV2(InvokeScriptTransactionDiff.scala:303)
	at com.wavesplatform.state.diffs.invoke.InvokeScriptTransactionDiff$.$anonfun$apply$4(InvokeScriptTransactionDiff.scala:103)
	at com.wavesplatform.metrics.TxProcessingStats$TxTimerExt$.measureForType$extension(TxProcessingStats.scala:26)
	at com.wavesplatform.state.diffs.invoke.InvokeScriptTransactionDiff$.executeMainScript$1(InvokeScriptTransactionDiff.scala:76)
	at com.wavesplatform.state.diffs.invoke.InvokeScriptTransactionDiff$.executeInvoke$1(InvokeScriptTransactionDiff.scala:133)
	at com.wavesplatform.state.diffs.invoke.InvokeScriptTransactionDiff$.$anonfun$apply$40(InvokeScriptTransactionDiff.scala:266)
	at com.wavesplatform.transaction.smart.script.trace.TracedResult.flatMap(TracedResult.scala:28)
	at com.wavesplatform.state.diffs.invoke.InvokeScriptTransactionDiff$.$anonfun$apply$30(InvokeScriptTransactionDiff.scala:225)
	at com.wavesplatform.transaction.smart.script.trace.TracedResult.flatMap(TracedResult.scala:28)
	at com.wavesplatform.state.diffs.invoke.InvokeScriptTransactionDiff$.$anonfun$apply$28(InvokeScriptTransactionDiff.scala:223)
	at com.wavesplatform.transaction.smart.script.trace.TracedResult.flatMap(TracedResult.scala:28)
	at com.wavesplatform.state.diffs.invoke.InvokeScriptTransactionDiff$.apply(InvokeScriptTransactionDiff.scala:222)
	at com.wavesplatform.state.diffs.TransactionDiffer$.$anonfun$transactionDiff$1(TransactionDiffer.scala:164)
	at com.wavesplatform.metrics.TxProcessingStats$TxTimerExt$.measureForType$extension(TxProcessingStats.scala:26)
	at com.wavesplatform.state.diffs.TransactionDiffer$.transactionDiff(TransactionDiffer.scala:161)
	at com.wavesplatform.state.diffs.TransactionDiffer$.$anonfun$validate$3(TransactionDiffer.scala:75)
	at com.wavesplatform.transaction.smart.script.trace.TracedResult.flatMap(TracedResult.scala:28)
	at com.wavesplatform.state.diffs.TransactionDiffer$.$anonfun$validate$2(TransactionDiffer.scala:74)
	at com.wavesplatform.transaction.smart.script.trace.TracedResult.flatMap(TracedResult.scala:28)
	at com.wavesplatform.state.diffs.TransactionDiffer$.$anonfun$validate$1(TransactionDiffer.scala:73)
	at com.wavesplatform.transaction.smart.script.trace.TracedResult.flatMap(TracedResult.scala:28)
	at com.wavesplatform.state.diffs.TransactionDiffer$.validate(TransactionDiffer.scala:72)
	at com.wavesplatform.state.diffs.TransactionDiffer$.apply(TransactionDiffer.scala:38)
	at com.wavesplatform.state.diffs.BlockDiffer$.$anonfun$apply$1(BlockDiffer.scala:152)
	at com.wavesplatform.state.diffs.BlockDiffer$.$anonfun$apply$2(BlockDiffer.scala:160)
	at scala.collection.IterableOnceOps.foldLeft(IterableOnce.scala:646)
	at scala.collection.IterableOnceOps.foldLeft$(IterableOnce.scala:642)
	at scala.collection.AbstractIterable.foldLeft(Iterable.scala:919)
	at com.wavesplatform.state.diffs.BlockDiffer$.apply(BlockDiffer.scala:156)
	at com.wavesplatform.state.diffs.BlockDiffer$.$anonfun$fromBlockTraced$12(BlockDiffer.scala:86)
	at com.wavesplatform.transaction.smart.script.trace.TracedResult.flatMap(TracedResult.scala:28)
	at com.wavesplatform.state.diffs.BlockDiffer$.fromBlockTraced(BlockDiffer.scala:77)
	at com.wavesplatform.state.diffs.BlockDiffer$.fromBlock(BlockDiffer.scala:38)
	at com.wavesplatform.state.BlockchainUpdaterImpl.$anonfun$processBlock$4(BlockchainUpdaterImpl.scala:222)
	at scala.util.Either.flatMap(Either.scala:352)
	at com.wavesplatform.state.BlockchainUpdaterImpl.$anonfun$processBlock$1(BlockchainUpdaterImpl.scala:201)
	at com.wavesplatform.state.BlockchainUpdaterImpl.inLock(BlockchainUpdaterImpl.scala:51)
	at com.wavesplatform.state.BlockchainUpdaterImpl.writeLock(BlockchainUpdaterImpl.scala:56)
	at com.wavesplatform.state.BlockchainUpdaterImpl.processBlock(BlockchainUpdaterImpl.scala:190)
	at com.wavesplatform.state.appender.package$.$anonfun$appendExtensionBlock$2(package.scala:78)
	at com.wavesplatform.metrics.package$TimerExt$.measureWithFilter$extension(package.scala:18)
	at com.wavesplatform.metrics.package$TimerExt$.measureSuccessful$extension(package.scala:11)
	at com.wavesplatform.state.appender.package$.$anonfun$appendExtensionBlock$1(package.scala:78)
	at scala.util.Either.flatMap(Either.scala:352)
	at com.wavesplatform.state.appender.package$.appendExtensionBlock(package.scala:77)
	at com.wavesplatform.state.appender.ExtensionAppender$.$anonfun$apply$10(ExtensionAppender.scala:60)
	at scala.collection.Iterator$$anon$9.next(Iterator.scala:575)
	at scala.collection.Iterator$$anon$16.next(Iterator.scala:823)
	at scala.collection.Iterator$$anon$16.next(Iterator.scala:818)
	at scala.collection.IterableOnceOps.collectFirst(IterableOnce.scala:1079)
	at scala.collection.IterableOnceOps.collectFirst$(IterableOnce.scala:1071)
	at scala.collection.AbstractIterable.collectFirst(Iterable.scala:919)
	at com.wavesplatform.state.appender.ExtensionAppender$.$anonfun$apply$9(ExtensionAppender.scala:66)
	at scala.util.Either.flatMap(Either.scala:352)
	at com.wavesplatform.state.appender.ExtensionAppender$.$anonfun$apply$3(ExtensionAppender.scala:55)
	at scala.util.Either.flatMap(Either.scala:352)
	at com.wavesplatform.state.appender.ExtensionAppender$.appendExtension$1(ExtensionAppender.scala:39)
	at com.wavesplatform.state.appender.ExtensionAppender$.$anonfun$apply$24(ExtensionAppender.scala:119)
	at map @ com.wavesplatform.state.appender.ExtensionAppender$.apply(ExtensionAppender.scala:119)
	at doOnFinish @ com.wavesplatform.Application.$anonfun$run$25(Application.scala:277)
	at asyncBoundary @ com.wavesplatform.network.RxExtensionLoader$.$anonfun$apply$28(RxExtensionLoader.scala:214)
	at map @ com.wavesplatform.network.RxExtensionLoader$.$anonfun$apply$28(RxExtensionLoader.scala:219)
```

**To Reproduce**
1. Just run the node
2. It gets stuck on block 2871360
3. And then crashes with the above error

**Expected behavior**
It shouldn't crash

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Version: v1.3.9

**Additional context**
https://wavesblockexplorer.com/blocks also seems to be stuck on the same block

**Comment by jaydp17** [comment]
This is fixed for us.

To the core contributors, I'm leaving this issue open so that the team at least has the chance to take a look at the issue. But please feel free to close this

**Comment by phearnot** [comment]
This issue has been fixed in [v1.3.10](https://github.com/wavesplatform/Waves/releases/tag/v1.3.10).
